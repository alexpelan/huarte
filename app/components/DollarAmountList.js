import React from 'react';
import {
  StatusBar,
  ListView,
  Text,
} from 'react-native';

import { selectQuestion } from '../actions/index';

import DailyDoubleBid from './DailyDoubleBid';
import Question from './Question';

import styles from '../styles/styles';

import StateHelper from '../util/StateHelper';

class DollarAmountList extends React.Component {
  constructor(props) {
    super(props);
    const dataSource = new ListView.DataSource({
      rowHasChanged: (row1, row2) => row1 !== row2,
    });

    this.state = {
      dataSource: dataSource.cloneWithRows(props.category.clues),
    };
  }

  selectClue = (clue, clueIndex) => {
    const store = this.props.store;
    const categoryName = this.props.category.name;
    store.dispatch(
      selectQuestion(
        StateHelper.getCurrentGame(store).currentRound,
        this.props.categoryIndex,
        clueIndex,
      ),
    );
    if (clue.isDailyDouble) {
      this.props.navigator.push({
        title: 'Daily Double!',
        component: DailyDoubleBid,
        passProps: {
          clue,
          clueIndex,
          store,
          categoryName,
          isSkippable: false,
        },
      });
    } else {
      this.props.navigator.push({
        title: clue.value,
        navigationBarHidden: true,
        component: Question,
        passProps: {
          clue,
          isSkippable: true,
          store,
          categoryName,
        },
      });
    }
  };

  renderClue = (clue, clueIndex) => (
    <Text
      style={[styles.listItem, clue.isCompleted && styles.listItemDisabled]}
      onPress={() => {
        if (!clue.isCompleted) {
          this.selectClue(clue, clueIndex);
        }
      }
      }
    >
      {clue.value}
    </Text>
  );

  render() {
    StatusBar.setBarStyle('default', true);
    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={(clue, sectionId, clueIndex) => this.renderClue(clue, clueIndex)}
        automaticallyAdjustContentInsets={false} // ????? https://github.com/facebook/react-native/issues/721
        style={styles.listView}
      />
    );
  }
}

export default DollarAmountList;
