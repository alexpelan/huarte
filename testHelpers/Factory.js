import { createStore, applyMiddleware } from "redux";
import thunkMiddleware from 'redux-thunk';

import huarteApp from "../app/reducers/huarteApp";

class Factory {
	static createNewStore() {
		return createStore(huarteApp, applyMiddleware(thunkMiddleware));
	}

	static game() {
		return {
				"_id": "58bc7c304a0eb9358ead01e3",
				"id": "5539",
				"displayName": "#7477, aired 2017-02-28",
				"season": "33",
				"show_number": "Show #7477",
				"air_date": "Tuesday, February 28, 2017",
				"jeopardy": {
					"categories": [
						{
							"name": "VALLEYS",
							"clues": [
								{
									"value": "$200",
									"question": "Today part of Israel, the Valley of Elah is where these 2 fought as recounted in the first book of Samuel",
									"answer": "David and Goliath (or the Philistines and the Israelites)",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$400",
									"question": "Lincoln signed the bill protecting this Calif. valley in 1864 & Benjamin Harrison signed the bill making it a natl. park in 1890",
									"answer": "Yosemite",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$600",
									"question": "The vast wheat fields of Canada's Manitoba province are found in the valley of this \"colorful\" river \"of the North\"",
									"answer": "the Red River",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "Once a national monument, this valley in Wyoming became part of Grand Teton National Park in 1950",
									"answer": "Jackson Hole",
									"media": [],
									"isDailyDouble": true
								}
							]
						},
						{
							"name": "BEASTLY MOVIES",
							"clues": [
								{
									"value": "$200",
									"question": "2000:\"____ Ugly\"",
									"answer": "coyote",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$400",
									"question": "2007: \"The Water ____: Legend of the Deep\"",
									"answer": "horse",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$600",
									"question": "1966: \"What's Up, ____ Lily?\"",
									"answer": "Tiger Lily",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "A 2003 animated film: \"Brother ____\"",
									"answer": "bear",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1000",
									"question": "2002:\"____ Rider\"",
									"answer": "whale",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "ABBREV.",
							"clues": [
								{
									"value": "$200",
									"question": "EP stands for this lofty position on a film or TV show",
									"answer": "executive producer",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$400",
									"question": "MFA is this graduate degree",
									"answer": "master of fine arts",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$600",
									"question": "The car automatically adjusts its speed with ACC, the adaptive type of this",
									"answer": "cruise control",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "The federal circuit has one of these intermediate bodies, C.O.A. for short",
									"answer": "court of appeals",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1000",
									"question": "G.I., this index, is the measure of how fast & how much a food raises blood sugar levels",
									"answer": "glycemic index",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "CROSSWORD CLUES \"Q\"",
							"clues": [
								{
									"value": "$200",
									"question": "Oddly charming(6)",
									"answer": "quaint",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$400",
									"question": "Those uneasy feelings(6)",
									"answer": "qualms (or queasy)",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$600",
									"question": "Edible clam type(6)",
									"answer": "quahog",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "A real quandary or a real bog(8)",
									"answer": "a quagmire",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1000",
									"question": "Serpent Toltec god(12)",
									"answer": "Quetzalcoatl",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "ROBOTS",
							"clues": [
								{
									"value": "$200",
									"question": "These 2 robots from a galaxy far, far away first appeared in a 1977 hit movie",
									"answer": "R2-D2 and C-3P0",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$400",
									"question": "Many computer tablets use this operating system created by Google & named for a humanoid robot",
									"answer": "Android",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$600",
									"question": "A design for a self-propelled robot vehicle was drawn up by this man around 1478; built in 2004, it worked",
									"answer": "da Vinci",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_J_14.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "Like its pal Opportunity, this robotic rover, launched into space June 10, 2003, has roved for miles on Mars",
									"answer": "Spirit",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1000",
									"question": "In \"I, Robot\", a sci-fi collection by this author, a robot become self-aware & begins to feel emotions",
									"answer": "(Isaac) Asimov",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "OLD STURBRIDGE VILLAGE",
							"clues": [
								{
									"value": "$200",
									"question": "(Sarah of the Clue Crew reports from Old Sturbridge Village, Massachusetts.)  A bustling place in the 18th century, used as everything from a parade ground to a town dump, by the early 1800s, the center of the typical New England village was becoming we think of as the peaceful manicured green, or as they call it in Boston, this",
									"answer": "the common",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_J_09.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$400",
									"question": "(Sarah of the Clue Crew reports from Old Sturbridge Village, Massachusetts.)  The miller made his living by keeping 1/16th of what he milled, as village residents would bring grain to be turned into flour; this word \"to the mill\", you might say",
									"answer": "grist",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_J_10.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$600",
									"question": "(Jimmy of the Clue Crew reports from Old Sturbridge Village, Massachusetts.)  1830s villagers, wanting personal protection, had the same main option as in Revolutionary War times--this type of gun named for the type of stone that sparks its firing mechanism",
									"answer": "flintlock",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_J_11.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "(Sarah of the Clue Crew reports from Old Sturbridge Village, Massachusetts.)  During the same era recreated at Old Sturbridge Village, this poet honored the honest sweat of the village blacksmith & wrote that children loved to see the flaming forge",
									"answer": "Henry Wadsworth Longfellow",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_J_12.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$1000",
									"question": "(Sarah of the Clue Crew reports from Old Sturbridge Village, Massachusetts.)  At Old Sturbridge Village, the meeting house is thought of as belonging to this denomination, a mouthful of a name the descendants of the Puritans adopted because it represented the local group of worshippers",
									"answer": "Congregationalist",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_J_25.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								}
							]
						}
					],
					"highestDollarAmount": 1000
				},
				"double_jeopardy": {
					"categories": [
						{
							"name": "ON YOUR LEFT!",
							"clues": [
								{
									"value": "$400",
									"question": "Left of Spain",
									"answer": "Portugal",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "Left of Argentina",
									"answer": "Chile",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "1980s BESTSELLERS",
							"clues": [
								{
									"value": "$400",
									"question": "In 1986 he celebrated his 82nd birthday with the publication of \"You're Only Old Once!\"",
									"answer": "Dr. Seuss",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "In this 1988 bestseller, James Michener told the story of the 49th state",
									"answer": "Alaska",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1200",
									"question": "In 1981 he checked into the bestseller list with \"The Hotel New Hampshire\"",
									"answer": "John Irving",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1600",
									"question": "His 1987 novel \"Texasville\" reintroduced the characters of \"The Last Picture Show\", all 30 years older",
									"answer": "Larry McMurtry",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$2000",
									"question": "The top fiction book of 1985 was Jean Auel's novel about these Ice Age \"Hunters\"",
									"answer": "The Mammoth Hunters",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "A SHOW OF HANS",
							"clues": [
								{
									"value": "$400",
									"question": "A statue of him in Central Park shows him reading from \"The Ugly Duckling\"",
									"answer": "Hans Christian Andersen",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_DJ_16.jpg",
											"type": "image"
										},
										{
											"url": "http://www.j-archive.com/media/2017-02-28_DJ_16a.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "This \"Die Hard\" character made AFI's list of the 100 greatest movie heroes & villains of all time",
									"answer": "Hans Gruber",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1200",
									"question": "We're counting on you to know that he gave his name to the device seen here",
									"answer": "Hans Geiger",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_DJ_25.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								},
								{
									"value": "$1600",
									"question": "He's composed music for more than 100 films, including \"The Dark Knight\" & \"Interstellar\"",
									"answer": "Hans Zimmer",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$2000",
									"question": "Hans Sachs was this type of crooner of Nuremberg, later a Wagner title character",
									"answer": "the Meistersinger",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_DJ_23.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "ALBUM COVERS",
							"clues": [
								{
									"value": "$400",
									"question": "A close-up of Springsteen's denim-clad backside in front of Old Glory",
									"answer": "Born in the U.S.A.",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "The pink Beverly Hills Hotel, shot at sunset",
									"answer": "Hotel California",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1200",
									"question": "A naked baby underwater reaches towards a dollar bill on a string",
									"answer": "Nevermind",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1600",
									"question": "On a black background, light goes through a prism & comes out a rainbow",
									"answer": "Dark Side of the Moon",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$2000",
									"question": "David Bowie sports a makeup lightning bolt across his face",
									"answer": "Aladdin Sane",
									"media": [],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "ANCIENT BATTLES",
							"clues": [
								{
									"value": "$400",
									"question": "In 66 A.D. victory by Jewish forces over a Roman army at Beth-Horon led to a brief liberation of this capital city",
									"answer": "Jerusalem",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "This commander's victory over Pompey in 48 B.C. at Pharsalus took him to the pinnacle of power",
									"answer": "Julius Caesar",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1200",
									"question": "7,000 Thebans at the Battle of Leuctra in 371 B.C. broke the power this Greek city had enjoyed for a generation",
									"answer": "Sparta",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1600",
									"question": "Defeat at the 207 B.C. Battle of the Metaurus River ended this man's efforts to conquer Italy",
									"answer": "Hannibal",
									"media": [],
									"isDailyDouble": true
								},
								{
									"value": "$2000",
									"question": "It was Cor vs. Cor when this most northerly of the Ionian islands took on Corinth in ancient Greece's first naval battle",
									"answer": "Corfu (or Corcyra)",
									"media": [
										{
											"url": "http://www.j-archive.com/media/2017-02-28_DJ_18a.jpg",
											"type": "image"
										},
										{
											"url": "http://www.j-archive.com/media/2017-02-28_DJ_18.jpg",
											"type": "image"
										}
									],
									"isDailyDouble": false
								}
							]
						},
						{
							"name": "TALK LIKE A CANADIAN",
							"clues": [
								{
									"value": "$400",
									"question": "A stagette party in Toronto is called this south of the border",
									"answer": "a bachelorette party",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$800",
									"question": "If it's cold in Calgary, put on your bunny hug, one of these items that are also popular Stateside",
									"answer": "a hoodie",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1200",
									"question": "When dining on poutine, keep handy une serviette, this",
									"answer": "a napkin",
									"media": [],
									"isDailyDouble": false
								},
								{
									"value": "$1600",
									"question": "Originally a derogatory term for Depression-era gasoline thieves, today it means any foolish or uncivilized person",
									"answer": "a hoser",
									"media": [],
									"isDailyDouble": true
								},
								{
									"value": "$2000",
									"question": "Got a Chesterfield inside that Regina mansion? It's one of these, not a cigarette",
									"answer": "a sofa",
									"media": [],
									"isDailyDouble": false
								}
							]
						}
					],
					"highestDollarAmount": 2000
				},
				"final_jeopardy": {
					"categories": [
						{
							"name": "RELIGION",
							"question": "Famous Catholics who've publicly answered this question include Susan Boyle (sweets) & Paul Ryan (beer)",
							"answer": "What did you give up for Lent?"
						}
					]
				}

		};
	}

	static gameList() {
		return {
			"status": "ok",
			"games": {
				"5403": {
					"displayName": "#7357, aired 2016-09-13"
				},
				"5404": {
					"displayName": "#7356, aired 2016-09-12"
				},
				"5405": {
					"displayName": "#7358, aired 2016-09-14"
				},
				"5406": {
					"displayName": "#7359, aired 2016-09-15"
				},
				"5407": {
					"displayName": "#7360, aired 2016-09-16"
				},
				"5408": {
					"displayName": "#7362, aired 2016-09-20"
				},
				"5409": {
					"displayName": "#7361, aired 2016-09-19"
				},
				"5410": {
					"displayName": "#7363, aired 2016-09-21"
				},
				"5411": {
					"displayName": "#7364, aired 2016-09-22"
				},
				"5412": {
					"displayName": "#7365, aired 2016-09-23"
				},
				"5413": {
					"displayName": "#7366, aired 2016-09-26"
				},
				"5415": {
					"displayName": "#7367, aired 2016-09-27"
				},
				"5418": {
					"displayName": "#7368, aired 2016-09-28"
				},
				"5419": {
					"displayName": "#7369, aired 2016-09-29"
				},
				"5421": {
					"displayName": "#7370, aired 2016-09-30"
				},
				"5422": {
					"displayName": "#7371, aired 2016-10-03"
				},
				"5423": {
					"displayName": "#7372, aired 2016-10-04"
				},
				"5424": {
					"displayName": "#7373, aired 2016-10-05"
				},
				"5426": {
					"displayName": "#7374, aired 2016-10-06"
				},
				"5427": {
					"displayName": "#7375, aired 2016-10-07"
				},
				"5428": {
					"displayName": "#7376, aired 2016-10-10"
				},
				"5429": {
					"displayName": "#7377, aired 2016-10-11"
				},
				"5430": {
					"displayName": "#7378, aired 2016-10-12"
				},
				"5431": {
					"displayName": "#7379, aired 2016-10-13"
				},
				"5432": {
					"displayName": "#7380, aired 2016-10-14"
				},
				"5433": {
					"displayName": "#7382, aired 2016-10-18"
				},
				"5437": {
					"displayName": "#7381, aired 2016-10-17"
				},
				"5438": {
					"displayName": "#7383, aired 2016-10-19"
				},
				"5439": {
					"displayName": "#7384, aired 2016-10-20"
				},
				"5440": {
					"displayName": "#7385, aired 2016-10-21"
				},
				"5441": {
					"displayName": "#7387, aired 2016-10-25"
				},
				"5442": {
					"displayName": "#7386, aired 2016-10-24"
				},
				"5443": {
					"displayName": "#7392, aired 2016-11-01"
				},
				"5444": {
					"displayName": "#7388, aired 2016-10-26"
				},
				"5445": {
					"displayName": "#7389, aired 2016-10-27"
				},
				"5446": {
					"displayName": "#7390, aired 2016-10-28"
				},
				"5447": {
					"displayName": "#7391, aired 2016-10-31"
				},
				"5448": {
					"displayName": "#7393, aired 2016-11-02"
				},
				"5449": {
					"displayName": "#7397, aired 2016-11-08"
				},
				"5451": {
					"displayName": "#7394, aired 2016-11-03"
				},
				"5452": {
					"displayName": "#7395, aired 2016-11-04"
				},
				"5453": {
					"displayName": "#7396, aired 2016-11-07"
				},
				"5454": {
					"displayName": "#7398, aired 2016-11-09"
				},
				"5455": {
					"displayName": "#7399, aired 2016-11-10"
				},
				"5456": {
					"displayName": "#7400, aired 2016-11-11"
				},
				"5457": {
					"displayName": "#7401, aired 2016-11-14"
				},
				"5458": {
					"displayName": "#7402, aired 2016-11-15"
				},
				"5459": {
					"displayName": "#7403, aired 2016-11-16"
				},
				"5460": {
					"displayName": "#7404, aired 2016-11-17"
				},
				"5461": {
					"displayName": "#7405, aired 2016-11-18"
				},
				"5462": {
					"displayName": "#7406, aired 2016-11-21"
				},
				"5463": {
					"displayName": "#7407, aired 2016-11-22"
				},
				"5465": {
					"displayName": "#7408, aired 2016-11-23"
				},
				"5466": {
					"displayName": "#7409, aired 2016-11-24"
				},
				"5467": {
					"displayName": "#7410, aired 2016-11-25"
				},
				"5468": {
					"displayName": "#7412, aired 2016-11-29"
				},
				"5469": {
					"displayName": "#7417, aired 2016-12-06"
				},
				"5470": {
					"displayName": "#7422, aired 2016-12-13"
				},
				"5471": {
					"displayName": "#7423, aired 2016-12-14"
				},
				"5472": {
					"displayName": "#7424, aired 2016-12-15"
				},
				"5473": {
					"displayName": "#7425, aired 2016-12-16"
				},
				"5474": {
					"displayName": "#7426, aired 2016-12-19"
				},
				"5475": {
					"displayName": "#7427, aired 2016-12-20"
				},
				"5476": {
					"displayName": "#7428, aired 2016-12-21"
				},
				"5477": {
					"displayName": "#7429, aired 2016-12-22"
				},
				"5478": {
					"displayName": "#7411, aired 2016-11-28"
				},
				"5479": {
					"displayName": "#7413, aired 2016-11-30"
				},
				"5480": {
					"displayName": "#7414, aired 2016-12-01"
				},
				"5481": {
					"displayName": "#7415, aired 2016-12-02"
				},
				"5482": {
					"displayName": "#7416, aired 2016-12-05"
				},
				"5483": {
					"displayName": "#7418, aired 2016-12-07"
				},
				"5484": {
					"displayName": "#7419, aired 2016-12-08"
				},
				"5485": {
					"displayName": "#7420, aired 2016-12-09"
				},
				"5486": {
					"displayName": "#7421, aired 2016-12-12"
				},
				"5487": {
					"displayName": "#7430, aired 2016-12-23"
				},
				"5488": {
					"displayName": "#7431, aired 2016-12-26"
				},
				"5489": {
					"displayName": "#7432, aired 2016-12-27"
				},
				"5490": {
					"displayName": "#7433, aired 2016-12-28"
				},
				"5491": {
					"displayName": "#7434, aired 2016-12-29"
				},
				"5492": {
					"displayName": "#7435, aired 2016-12-30"
				},
				"5493": {
					"displayName": "#7436, aired 2017-01-02"
				},
				"5494": {
					"displayName": "#7437, aired 2017-01-03"
				},
				"5495": {
					"displayName": "#7438, aired 2017-01-04"
				},
				"5496": {
					"displayName": "#7439, aired 2017-01-05"
				},
				"5497": {
					"displayName": "#7440, aired 2017-01-06"
				},
				"5498": {
					"displayName": "#7441, aired 2017-01-09"
				},
				"5499": {
					"displayName": "#7442, aired 2017-01-10"
				},
				"5500": {
					"displayName": "#7443, aired 2017-01-11"
				},
				"5501": {
					"displayName": "#7444, aired 2017-01-12"
				},
				"5503": {
					"displayName": "#7445, aired 2017-01-13"
				},
				"5504": {
					"displayName": "#7446, aired 2017-01-16"
				},
				"5505": {
					"displayName": "#7447, aired 2017-01-17"
				},
				"5506": {
					"displayName": "#7448, aired 2017-01-18"
				},
				"5507": {
					"displayName": "#7449, aired 2017-01-19"
				},
				"5508": {
					"displayName": "#7450, aired 2017-01-20"
				},
				"5509": {
					"displayName": "#7451, aired 2017-01-23"
				},
				"5510": {
					"displayName": "#7452, aired 2017-01-24"
				},
				"5511": {
					"displayName": "#7453, aired 2017-01-25"
				},
				"5512": {
					"displayName": "#7457, aired 2017-01-31"
				},
				"5513": {
					"displayName": "#7456, aired 2017-01-30"
				},
				"5514": {
					"displayName": "#7458, aired 2017-02-01"
				},
				"5515": {
					"displayName": "#7459, aired 2017-02-02"
				},
				"5516": {
					"displayName": "#7460, aired 2017-02-03"
				},
				"5517": {
					"displayName": "#7462, aired 2017-02-07"
				},
				"5518": {
					"displayName": "#7461, aired 2017-02-06"
				},
				"5519": {
					"displayName": "#7463, aired 2017-02-08"
				},
				"5520": {
					"displayName": "#7464, aired 2017-02-09"
				},
				"5521": {
					"displayName": "#7465, aired 2017-02-10"
				},
				"5522": {
					"displayName": "#7467, aired 2017-02-14"
				},
				"5523": {
					"displayName": "#7466, aired 2017-02-13"
				},
				"5524": {
					"displayName": "#7468, aired 2017-02-15"
				},
				"5526": {
					"displayName": "#7469, aired 2017-02-16"
				},
				"5527": {
					"displayName": "#7470, aired 2017-02-17"
				},
				"5529": {
					"displayName": "#7471, aired 2017-02-20"
				},
				"5530": {
					"displayName": "#7472, aired 2017-02-21"
				},
				"5531": {
					"displayName": "#7473, aired 2017-02-22"
				},
				"5532": {
					"displayName": "#7474, aired 2017-02-23"
				},
				"5533": {
					"displayName": "#7475, aired 2017-02-24"
				},
				"5534": {
					"displayName": "#7454, aired 2017-01-26"
				},
				"5535": {
					"displayName": "#7455, aired 2017-01-27"
				},
				"5539": {
					"displayName": "#7477, aired 2017-02-28"
				}
			}
		};
	}

	static seasonList() {
		return {
			"status": "ok",
			"seasons": [
				{
					"_id": "58bc7c304a0eb9358ead01c4",
					"id": "29",
					"displayName": "Season 29",
					"sortNumber": 4,
					"seasonId": "29"
				},
				{
					"_id": "58bc7c304a0eb9358ead01c3",
					"id": "30",
					"displayName": "Season 30",
					"sortNumber": 3,
					"seasonId": "30"
				},
				{
					"_id": "58bc7c304a0eb9358ead01c2",
					"id": "31",
					"displayName": "Season 31",
					"sortNumber": 2,
					"seasonId": "31"
				},
				{
					"_id": "58bc7c304a0eb9358ead01c1",
					"id": "32",
					"displayName": "Season 32",
					"sortNumber": 1,
					"seasonId": "32"
				},
				{
					"_id": "58bc7c304a0eb9358ead01c0",
					"id": "33",
					"displayName": "Season 33",
					"sortNumber": 0,
					"seasonId": "33"
				}
			]
		};
	}

}

export default Factory;