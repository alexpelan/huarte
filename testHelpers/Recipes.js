import { receiveSeasons, receiveGameList, receiveGame, selectGame } from '../app/actions/index';
import Factory from './Factory';

class Recipes {
  static loadSingleGame(store) {
    const seasonList = Factory.seasonList();
    const gameList = Factory.gameList();
    store.dispatch(receiveSeasons(seasonList));
    store.dispatch(receiveGameList(gameList, '33'));
    const game = Factory.game();
    store.dispatch(selectGame(game));
    store.dispatch(receiveGame(game, game.id));
    return game;
  }
}

export default Recipes;
