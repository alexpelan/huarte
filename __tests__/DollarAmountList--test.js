import 'react-native';
import React from 'react';
// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';
import DollarAmountList from '../app/components/DollarAmountList';
import Factory from '../testHelpers/Factory';
import Recipes from '../testHelpers/Recipes';

test('renders correctly', () => {
  const store = Factory.createNewStore();
  const game = Recipes.loadSingleGame(store);

  const tree = renderer.create(
    <DollarAmountList store={store} category={game.jeopardy.categories[0]} categoryIndex={0} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});


test('renders correctly with category comments (#5)', () => {
  const store = Factory.createNewStore();
  const game = Recipes.loadSingleGame(store);
  game.jeopardy.categories[0].categoryComments = 'Some helpful text since the clues might make no sense out of context.';

  const tree = renderer.create(
    <DollarAmountList store={store} category={game.jeopardy.categories[0]} categoryIndex={0} />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
