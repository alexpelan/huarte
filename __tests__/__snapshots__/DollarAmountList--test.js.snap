// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  dataSource={
    ListViewDataSource {
      "items": 4,
    }
  }
  renderFooter={[Function]}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "backgroundColor": "#0000af",
      "paddingTop": 70,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $200
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $400
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $600
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $800
    </Text>
  </View>
</RCTScrollView>
`;

exports[`renders correctly with category comments (#5) 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  dataSource={
    ListViewDataSource {
      "items": 4,
    }
  }
  renderFooter={[Function]}
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "backgroundColor": "#0000af",
      "paddingTop": 70,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $200
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $400
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $600
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Array [
          Object {
            "color": "white",
            "fontSize": 20,
            "marginBottom": 8,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      $800
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "white",
          "fontSize": 18,
          "textAlign": "center",
        }
      }
    >
      Some helpful text since the clues might make no sense out of context.
    </Text>
  </View>
</RCTScrollView>
`;
